### PR TITLE
Resolve JSON pointers in swagger spec

### PR DESCRIPTION
**Pull request by patrickw276**
Add support for resolving local file references in swagger specs (#254).

Edit: I'm having some trouble getting the build to pass... I think it's just isort complaining about import order. I ran isort on the two modules in my PR and it's still not happy.

### CODE REVIEW

**Comment by hjacobs** [comment]
@patrickw276 thanks for the PR! We'll look into it...

File: connexion/swagger.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
+"""
```
Comment (hjacobs) [inline connexion/swagger.py]: you can remove the license header as the `LICENSE` file is enough (and we should probably remove all existing file headers in a later PR)

**Comment by patrickw276** [comment]
I just realized that this code creates recursive dictionaries when resolving circular pointers contained in any file except the root. These recursive dictionaries will crash the swagger validator. My use case doesn't include circular references but I guess it's feasible that someones will. I don't mind fixing this but it may take me some time before I can get around to it. 

Would it be acceptable to raise an exception stating that circular references outside of root are not currently supported?

**Comment by jfinkhaeuser** [comment]
One of the annoying things with this is that the jsonschema validator underlying the validation does in fact (partially) resolve references.

I've got a parser that resolves references. I have this in an as-yet unreleased package related to API testing. I could factor it into its own package, if you'd like to use it.

**Comment by patrickw276** [comment]
@jfinkhaeuser Yeah, I knew going into writing this code that the jsonschema validator resolves references but I don't think the swagger spec validator does. Do you think a better approach would be leaving the references in and having the various libraries resolve them on their own? It seems like at some point the schema will need to be resolved into a single dictionary for connexion to pass around for routing, etc...

Is your parser on github somewhere? I'd be happy to take a look at it/potentially use it.

**Comment by jfinkhaeuser** [comment]
I'll make it available in the next few days, then, and link it here. It's on github, but the repo is private for the moment - I need to move it, package it, etc.

**Comment by jfinkhaeuser** [comment]
Try https://pypi.python.org/pypi/prance/0.1.0 and/or https://github.com/jfinkhaeuser/prance

It might not be what you're looking for. If you have additional test cases you need to work, let me know.

**Comment by patrickw276** [comment]
After more thought, I don't like my approach in this PR. I looked more into swagger UI and it can resolve relative references. I'm going to see what it would take to resolves references as needed and not bundle everything into a single schema on load.

**Comment by jfinkhaeuser** [comment]
IMHO the problem with resolving references as needed is that for validation, you need to resolve them anyway. If you want to validate on load, then you have to resolve on load. If you want to validate at run-time later on, then of course resolving later is just fine.

That's just my 2c, of course.

**Comment by patrickw276** [comment]
@jfinkhaeuser I'll have to dig into it a bit more but I think resolving on load would be better/safer. There's a good chance a different approach will use a lot of the same code (or your library, ty for making it available btw). Once I realized that the swagger UI supports relative references, bundling all documents into one seemed like too hacky of a solution.

**Pull request closed without merging**